cmd/darvaza/serve.go

```diff
@@ -24,21 +19,23 @@ var serveCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		// server is just a slice of things implementing
 		// the pr{oxy} interface
-		var server []pr
+		var server []types.Pr
 		for i := range cfg.Proxies {
 			if z := cfg.Proxies[i].New(); z != nil {
 				server = append(server, z)
 			}
 		}
-		var wg sync.WaitGroup
-		for i := range server {
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-				server[i].Run()
-			}(i)
-		}
-		wg.Wait()
+		go func() {
```

Comment:
aren't we missing a way to signal the for-select all proxies are finished?

Reply:
we are.

Reply:
the classic `defer close(done)` not good enough?

Reply:
> aren't we missing a way to signal the for-select all proxies are finished?

finished in what sense? ready for accepting connections or finished as in shutting down...

Reply:
I meant finished/done as no Runner is still Run()ing.
Ready() on tableflip terms is a different story

Reply:
fixed in de34958519060e68101e545542d2b6a451f4d99f